src/legacy/core_plugins/vis_default_editor/public/components/agg_params_helper.ts

```diff
@@ -63,19 +72,26 @@ function getAggParamsToRender({ agg, editorConfig, metricAggs, state }: ParamIns
         .filter((param: AggParam) => !get(editorConfig, [param.name, 'hidden'], false))) ||
     [];
 
+  const schema = getSchemaByName(schemas, agg.schema);
   // build collection of agg params components
   paramsToRender.forEach((param: AggParam, index: number) => {
     let indexedFields: ComboBoxGroupedOptions<IndexPatternField> = [];
     let fields: IndexPatternField[];
 
-    if (agg.schema.hideCustomLabel && param.name === 'customLabel') {
+    if (!schema || (schema.hideCustomLabel && param.name === 'customLabel')) {
       return;
     }
     // if field param exists, compute allowed fields
     if (param.type === 'field') {
-      const availableFields: IndexPatternField[] = (param as IFieldParamType).getAvailableFields(
-        agg
-      );
+      let availableFields: IndexPatternField[] = (param as IFieldParamType).getAvailableFields(agg);
+      // should be refactored in the future to provide a more general way
+      // for visualization to override some agg config settings
+      if (agg.type.name === 'top_hits') {
+        const allowStrings = _.get(schema, `aggSettings[${agg.type.name}].allowStrings`, false);
```

Comment:
handling special case with top_hits, where we want to limit allowed fields in most visualizations

Reply:
This again breaks the current behavior, I think the 4th time on my mind. (worth functional tests)
In the top hit aggregation we have another one param with type `field` - the `Sort on`.
In this case, you limit other fields, except the numeric only, since you allowed it in config.
Please, address this pr https://github.com/elastic/kibana/pull/56367 to look at how field filtering should work.

![top_hit_sort_on](https://user-images.githubusercontent.com/31325372/75960185-aa348a80-5ed0-11ea-9fa8-73f4df180926.PNG)